actionview/test/template/dependency_tracker_test.rb

```diff
@@ -243,6 +243,18 @@ class ERBTrackerTest < ActiveSupport::TestCase
   def make_tracker(name, template, view_paths = nil)
     ActionView::DependencyTracker::ERBTracker.new(name, template, view_paths)
   end
+
+  def test_dependencies_with_interpolation_non_trailing
```

Comment:
Can you move this up to the `SharedTrackerTests`? We want to make sure every Tracker behaves the same

Edit: actually, the current patch makes the ErbTracker behave differently than the others. Here's a fix for that: https://github.com/skipkayhil/rails/commit/7f87cf406d3aa34930e62af157c1ea72a3cda318

Reply:
Thanks @skipkayhil!
I cherry-picked your commit and reverted mine and co-authored: you did all the work 😅.